src/emitters/directional.cpp

```diff
@@ -59,7 +60,8 @@ class DirectionalEmitter final : public Emitter<Float, Spectrum> {
         } else if (props.has_property("to_world")) {
             m_world_transform = props.animated_transform("to_world");
         } else {
-            m_world_transform = new AnimatedTransform(ScalarTransform4f::scale(1.));
+            m_world_transform =
```

Comment:
I just realized that the `Endpoint` constructor already initializes the `m_wolrd_transform` transform if provided. So you original code was correct.